File: test/Video/test03.c
#include <libming.h>
 
int main (int argc, char **argv)
{
  SWFMovie movie;
  SWFVideoStream video;
  SWFDisplayItem item;
  SWFAction action;
 
  movie = newSWFMovieWithVersion(7);
  SWFMovie_setRate (movie, 1);
  SWFMovie_setDimension (movie, 200, 150);
  video = newSWFVideoStream ();
  SWFVideoStream_setDimension (video, 200, 150);
  item = SWFMovie_add (movie, (SWFBlock) video);
  SWFDisplayItem_setName (item, "video");
  action = newSWFAction (""
      "nc = new NetConnection ();"
      "nc.connect (null);"
      "ns = new NetStream (nc);"
      "video.attachVideo (ns);"
      "ns.play (\"video.flv\");"
      "");
  SWFMovie_add (movie, (SWFBlock) action);
  SWFMovie_save (movie, "test03.swf");
  return 0;
}

File: test/Video/test03.pl
#!/usr/bin/perl -w

use SWF qw(:ALL);

die "Usage: ".$0." <sourcedir>\n" unless @ARGV;

$m = SWF::Movie::newSWFMovieWithVersion(7);
$m->setRate(1);
$m->setDimension (200, 150);

$stream1 = new SWF::VideoStream();
$stream1->setDimension(200, 150);
$it = $m->add($stream1);
$it->setName("video");

$action = new SWF::Action("
	nc = new NetConnection ();
	nc.connect (null);
	ns = new NetStream (nc);
	video.attachVideo (ns);
	ns.play (\"video.flv\");
");
$m->add($action);

$m->save("test03.swf");


